tomcat-8.5/src/org/dogtagpki/tomcat/JSSUtil.java

```diff
@@ -46,20 +47,25 @@
     public static Log logger = LogFactory.getLog(JSSUtil.class);
 
     private String keyAlias;
-    private Set<String> protocols;
-    private Set<String> ciphers;
+    private static Set<String> protocols;
+    private static Set<String> ciphers;
+
+    static {
+        JSSEngineReferenceImpl eng = new JSSEngineReferenceImpl();
+        protocols = new HashSet<String>(Arrays.asList(eng.getSupportedProtocols()));
+        protocols = Collections.unmodifiableSet(protocols);
+        ciphers = new HashSet<String>(Arrays.asList(eng.getSupportedCipherSuites()));
+        ciphers = Collections.unmodifiableSet(ciphers);
+    }
```

Comment:
Would something like this work?
```
public class JSSUtil extends SSLUtilBase {

    private JSSEngine engine = new JSSEngineReferenceImpl();

    private Set<String> protocols = Collections.unmodifiableSet(
        new HashSet<String>(Arrays.asList(engine.getSupportedProtocols()))
    );

    public JSSUtil(SSLHostConfigCertificate cert) {
        super(cert);
        ...
    }
}
```
In general I'd rather avoid a static code block, but if that's not possible this is OK.

Reply:
Sure, that works. I just left them as static though.